### ISSUE TITLE
Verfication grid problems after server move - Records not being updated?

### ISSUE THREAD

**Issue by MNSmith** [BiologicalRecordsCentre/iRecord#671]
Now that iRecord is available again after the server move, I'm seeing issues with the Verification grid - Accepting or Querying records seems to have no effect on the displayed record, they remain on the grid with no changes apparent.

There are other comments on other problems on the iRecord forums.

**Comment by BirenRathod** [comment]
@johnvanbreda & @burkmarr 
Could you check this and let me know if this works ok,

**Comment by burkmarr** [comment]
Yes I think that saving a filte is working now.

**Comment by johnvanbreda** [comment]
I can still get problems on verification. If I go to the verification page without the www in the link (which still works) then the ajaxproxy returns:
{"error":"iform_ajaxproxy Error: Currently not logged in, and does not have permission to save data via the proxy."}
That's because it goes to the link with www in now. 

Don't we need to redirect brc.ac.uk/irecord* to www.brc.ac.uk/irecord so that all users are forced to use www (or forced not to if you prefer)?

**Comment by BirenRathod** [comment]
@johnvanbreda 
redirection already done. Please clear the cookies.

Also I found this error in log relate to verification....
PDOException: SQLSTATE[HY000]: General error: 1525 Incorrect DATETIME value: 'NULL': SELECT field_data_field_indicia_ae_due_date0.entity_type AS entity_type, field_data_field_indicia_ae_due_date0.entity_id AS entity_id, field_data_field_indicia_ae_due_date0.revision_id AS revision_id, field_data_field_indicia_ae_due_date0.bundle AS bundle FROM {field_data_field_indicia_ae_due_date} field_data_field_indicia_ae_due_date

**Comment by johnvanbreda** [comment]
@BirenRathod I've cleared my cookies but redirection is still not working for me. Also see my email about other problems that are now occurring. I've cleared both the Indicia and Drupal caches.

**Comment by kitenetter** [comment]
@johnvanbreda my new verification page doesn't seem to be showing any records added after 10 Sept (the old page does show records added on subsequent days). Also just had an error message "Elasticsearch query failed".

**Comment by johnvanbreda** [comment]
@kitenetter I've restarted a complete refresh of the Elasticsearch data this morning so we pick up all the changes to UKSI. It will probably be tomorrow before it is fully back to normal with new records arriving being added straight into ES.

**Comment by johnvanbreda** [comment]
@kitenetter - should be OK now?

**Comment by kitenetter** [comment]
@johnvanbreda yes, new records are showing again, thanks. I think we can close this one.

**Comment by kitenetter** [comment]
Re-opening: verifier John Tilt (warehouse user ID 18937) reports that his new verification page is not showing any pending records, while there are lots on the old page. I can reproduce that if I masquerade as him.

**Comment by johnvanbreda** [comment]
It was a problem with indexing against certain locations not being copied into Elasticsearch, if the location had a null code value. I've fixed this and triggered a re-index so John's new verification page will fill up over the next day or two.

**Comment by kitenetter** [comment]
@johnvanbreda I'm getting reports from a few verifiers that records are not appearing promptly in the new verification page. I'm not sure if some of this is just the normal lag in records getting copied into the elastic dataset - how long should that normally take?

Examples:
- John Tilt has over 1,000 pending records up to 09:22 on 27 Sep on the old page, but on the new page there are no pending records displayed - are we still waiting for the location indexing to work its way through the system?
- My Berkshire moths filter has records up to 17:08 on 26 Sep on the old page, but on the new page the latest is 12:35 on 23 Sep
- Stuart Roberts (using Hymenoptera - aculeates filter) has records up to 08:49 on 27 Sep on the old page, but on the new page the latest is 11:38 on 25 Sep

**Comment by johnvanbreda** [comment]
@biren - copying you in as there was an error on the warehouse which might need you to look into it.

@kitenetter - here's an explanation for you:
1. I triggered a complete re-index on the 25th. 
2. At 17:40 on the 25th (Biren - check the warehouse logs) there was an error relating to the PostgreSQL database: "server closed the connection unexpectedly". This might be a server issue rather than a code one.
3. Logstash fell over at that point and stopped doing any more records.

I've restarted Logstash and told it to process the record changes since the 25th first, then I'll reset it back again so it picks up the remaining records for re-indexing the BC branch data.

**Comment by BirenRathod** [comment]
@johnvanbreda 
I found that the server was restarted due to Sophos security patch update.

**Comment by johnvanbreda** [comment]
Note to self:
* BC boundaries were updated in locations.yml file a couple of weeks ago, so I've triggered re-index.
* Taxa which don't map to the UKSI list via TVK (e.g. recording aggregates in UKBMS) were not appearing on  Elasticsearch. I've changed the reports to include basic taxon info for ES in that case.

May just need to double check comparisons between 2 verification pages once re-index done before this can be closed.

**Comment by kitenetter** [comment]
Review if fixed following update

**Comment by kitenetter** [comment]
John Tilt has again reported that he has c. 700 records pending on his old verification page that do not appear on the new one, and I can reproduce that if I masquerade as him. 

I've tried looking for a couple of records:
- occurrence ID 11675134 originates from iRecord general data - I can't find it at all via the new verification page, even if filtering for all records
- occurrence ID 11583076 originates from iRecord app - also not visible via the new verification page

Most of the "missing" records appear to have been entered during August 2019.

**Comment by kitenetter** [comment]
Update at 7 Jan 2020:
- Old verification page, pending records = 852
- New verification page, pending records = 91
- Old verification page, accepted records = not displaying
- New verification page, accepted records = 154,998


- occurrence ID 11675134 is still unavailable on the new verification page, even if filtering for all records
- occurrence ID 11583076 is also not visible via the new verification page

So we still have a problem to resolve here.

**Comment by DavidRoy** [comment]
Review after refresh of data into ES index

**Comment by kitenetter** [comment]
Re-check John Tilt's view

**Comment by kitenetter** [comment]
Masqueraded as John Tilt as 12:13 on 6 April 2020.

Old verification page:
813 records up to 13153325
813 records from 10708898 to 13153325

Occurrence ID 11675134 now visible on new page.

Appears to be resolved, closing.
